com.unity.shadergraph/Editor/Data/Nodes/Input/CustomInterpolatorNode.cs

```diff
@@ -0,0 +1,278 @@
+using System;
+using System.Linq;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEditor.Graphing;
+using UnityEditor.ShaderGraph.Internal;
+using UnityEditor.Rendering;
+
+namespace UnityEditor.ShaderGraph
+{
+    [Serializable]
```

Comment:
Maybe stupid question, but why serializable is needed here? I saw most of our nodes do not have it

Reply:
Various nodes serialize their own data. Block deserialization happens in a post-deserialization step after nodes are deserialized. Custom Interpolator Nodes rely on weak references to their respective Custom Block Node, which may not be valid if the Custom Block Node doesn't exist or it hasn't yet loaded. This serialization data is just to ensure it loads correctly and doesn't cause downstream unexpected impact on ports that require concretization (eg. dynamic vectors).